### PR TITLE
feat(PasswordInput): add typescript types to PasswordInput

### LINKED ISSUE #12550: Add TypeScript types to PasswordInput

### PR DESCRIPTION
**Pull request by shryoo-ibm**
Closes #12550 

Part of #12513, Adding TypeScript types to carbon react components

#### Changelog

**Changed**

- Adds type declarations to the props of the PasswordInput component

#### Testing / Reviewing

Types can be tested by making a <PasswordInput /> component and trying out the different property values.

### CODE REVIEW

**Comment by shryoo-ibm** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/TextInput/PasswordInput.tsx [hunk 1]
```diff
@@ -308,7 +468,7 @@ PasswordInput.propTypes = {
    */
   size: FeatureFlags.enabled('enable-v11-release')
     ? PropTypes.oneOf(['sm', 'md', 'lg'])
-    : PropTypes.string,
+    : PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
```
Comment (tay1orjones) [inline packages/react/src/components/TextInput/PasswordInput.tsx]: This feature flag will always `true` now that we've released v11. This toggle could be removed entirely. The type definition should be updated as well. `xl` is no longer a valid value.

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Just one suggestion, otherwise I think this looks good!

File: packages/react/src/components/TextInput/PasswordInput.tsx (hunk 1 above)
Comment (shryoo-ibm) [reply packages/react/src/components/TextInput/PasswordInput.tsx]: Got it, I was a bit confused about that so the clarification is a big help!

**Review by tay1orjones — APPROVED** [review]
Looks great, thanks! 🎉

**Comment by tay1orjones** [comment]
@shryoo-ibm looks like there's a minor conflict to resolve - just regen the allcontributors probably

**Review by alisonjoseph — APPROVED** [review]

**Comment by aledavila** [comment]
bump @shryoo-ibm looks like you have some minor conflicts to fix :) then we can review thanks!

**Review by tay1orjones — APPROVED** [review]
Looks good to me 👍 thanks!

**Pull request merged**